zrml/swaps/src/migrations/_0_1_2_add_pool_market_id.rs

```diff
@@ -0,0 +1,110 @@
+use crate::{BalanceOf, Config, Pallet, Pools};
+use alloc::{collections::BTreeMap, vec::Vec};
+use frame_support::{
+    dispatch::Weight,
+    traits::{Get, GetPalletVersion, PalletVersion},
+};
+use zeitgeist_primitives::types::{Asset, Pool, PoolStatus};
+
+type PreviousPoolTy<T> = PreviousPool<BalanceOf<T>, <T as crate::Config>::MarketId>;
+
+#[derive(
+    Clone,
+    Eq,
+    PartialEq,
+    parity_scale_codec::Decode,
+    parity_scale_codec::Encode,
+    sp_runtime::RuntimeDebug,
+)]
+struct PreviousPool<B, MI> {
+    assets: Vec<Asset<MI>>,
+    pool_status: PoolStatus,
+    swap_fee: B,
+    total_weight: u128,
+    weights: BTreeMap<Asset<MI>, u128>,
+}
+
+pub fn migrate<T>() -> Weight
+where
+    T: Config,
+{
+    let mut weight: Weight = 0;
+    let pallet_version = PalletVersion {
+        major: 0,
+        minor: 1,
+        patch: 1,
+    };
+    let storage_version = <Pallet<T>>::storage_version().unwrap_or(pallet_version);
+
+    if storage_version == pallet_version {
+        let _ = <Pools<T>>::translate::<Option<PreviousPoolTy<T>>, _>(|_, pool_opt| {
+            weight = weight.saturating_add(T::DbWeight::get().reads(1));
+            if let Some(pool) = pool_opt {
+                weight = weight.saturating_add(T::DbWeight::get().writes(1));
+                Some(Some(Pool {
+                    assets: pool.assets,
+                    market_id: T::MarketId::from(0u8),
```

Comment:
The old pools will have an invalid associated `market_id`, what does this imply?

Reply:
The Swaps pallet is not aware of anything regarding markets, in fact, PM uses Swaps and not the other way around so there is no place to take market information from Swaps. This migration is just a way to enforce backwards compatibility.

Once #167 is merged, common logic between markets, pools, liquidity mining and court will be used inside `market-commons` so another migration must be created afterwards to correctly map swaps <-> migrations